src/components/productList/ProductList.component.tsx

```diff
@@ -1,24 +1,119 @@
+import { useEffect, useState } from 'react';
+
+import type { Category } from '@/components/filters/Filters.component.tsx';
+import FiltersComponent from '@/components/filters/Filters.component.tsx';
+import { SortOption } from '@/components/filters/Filters.component.tsx';
 import type { Product } from '@/interfaces/Products.ts';
 import mockProducts from '@/mocks/mockData.json';
 
+import PaginationComponent from '../pagination/Pagination.component.tsx';
+
 import ProductCard from './ProductCard.component.tsx';
 
 import styles from './ProductList.module.css';
 
-interface ProductsListProps {
-    addToCart: () => void;
-    theme: 'light' | 'dark';
-}
-const ProductsList: React.FC<ProductsListProps> = ({ addToCart, theme }) => (
-    <>
-        <div className={styles.container}>
-            <div className={styles.cardsPlacement}>
-                {mockProducts.map((product: Product) => (
-                    <ProductCard key={product.id} product={product} addToCart={addToCart} theme={theme} />
-                ))}
+const ProductsList: React.FC = () => {
+    const [currentPage, setCurrentPage] = useState(1);
+    const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
+    const [selectedCategories, setSelectedCategories] = useState<Category[]>([]);
+    const [sortOption, setSortOption] = useState<SortOption>(SortOption.PRICE_HIGH_TO_LOW);
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [searchInput, setSearchInput] = useState<string>('');
+
+    const productsPerPage = 8;
+
+    useEffect(() => {
+        let updatedProducts = mockProducts;
+
+        const filterProducts = () => {
+            if (selectedCategories.length > 0) {
+                updatedProducts = updatedProducts.filter((product) => selectedCategories.includes(product.category.name as Category));
+            }
+
+            if (searchQuery) {
+                updatedProducts = updatedProducts.filter((product) => product.title.toLowerCase().includes(searchQuery.toLowerCase()));
+            }
+        };
+        filterProducts();
+
+        const sortProducts = () => {
+            switch (sortOption) {
+                case SortOption.PRICE_LOW_TO_HIGH: {
+                    updatedProducts = updatedProducts.sort((a, b) => a.price - b.price);
```

Comment:
React state must be immutable 
`sort` mutates the array. You have to use the immutable `toSorted`  
You can learn more here: https://dev.to/endeavourmonk/why-we-should-not-mutate-in-react-18ej